doc/developer/design/20230421_stabilize_monotonic_select.md

```diff
@@ -100,21 +100,25 @@ There are two strategies that we propose to lower the expense of monotonic opera
 
-The full set of single-time monotonicity enforcers is: `Plan::Reduce`, `Plan::TopK`, `Plan::Threshold`, `Plan::ArrangeBy`, `Plan::Constant` iff the constant is an `EvalError` or all its rows have `Diff` values greater than zero, and `Plan::Get` iff one of the following conditions is true: (1) the correspoding `GetPlan` is either `GetPlan::PassArrangements` or `GetPlan::Arrangement`; (2) The corresponding `GetPlan` is `GetPlan::Collection`, the `id` in `Plan::Get` is a global ID for a monotonic source or for an index.
+The logic in the analysis above is similar to what is implemented at MIR level in the `MonotonicFlag` analysis. However, we consider only single-time contexts and a different level of abstraction, since the analysis is performed in LIR.
 
-For the purposes of this proposal, we consider the full set of single-time monotonicity breakers to be: `Plan::Negate`, `Plan::Constant` iff the constant contains rows with non-positive `Diff` values, and `Plan::Get` nodes iff the corresponding `GetPlan` is `GetPlan::Collection` and the `id` in `Plan::Get` is either a local ID or a global ID for a non-monotonic source. Note that including local IDs in the set of single-time monotonicity breakers is imprecise: The local ID could in fact be produced by a plan subgraph that would result in physical monotonicity in a single-time context. We believe this conservative approximation is sufficient for most one-shot queries, and avoids the complexity of needing to perform a complete analysis of `Plan::Let` nodes to enrich local IDs with single-time physical monotonicity information.
+The full set of single-time monotonicity enforcers is: `Plan::Reduce`, `Plan::TopK`, `Plan::Threshold`, `Plan::ArrangeBy`, `Plan::Constant` iff the constant is an `EvalError` or all its rows have `Diff` values greater than zero, and `Plan::Get` iff one of the following conditions is true: (1) the correspoding `GetPlan` is either `GetPlan::PassArrangements` or `GetPlan::Arrangement`; (2) The corresponding `GetPlan` is `GetPlan::Collection`, the `id` in `Plan::Get` is a local or global ID for a monotonic source or for an index.
 
-All other `Plan` nodes or variations not covered are monotonicity keepers.
+For monotonicity breakers, we note that any `Plan` nodes contained in the `values` of a `Plan::LetRec` expression are not in a single-time context due to iteration. This is why `refine_single_time_operator_selection` only considers the `body` of a `Plan::LetRec`, explicitly excluding traversal of `values`. So we must be careful in the analysis to consider the bindings introduced by `Plan::LetRec` similarly to non-monotonic sources. Additionally, monotonicity breakers comprise `Plan::Negate`, `Plan::Constant` iff the constant contains rows with non-positive `Diff` values, and `Plan::Get` nodes iff the corresponding `GetPlan` is `GetPlan::Collection` and the `id` in `Plan::Get` is either a local or global ID for a non-monotonic source.
```

Comment:
nit: "similarly to non-monotonic sources" ⇒ "to be monotonicity breakers" (less value).

Reply:
Fixed!